icepyx/core/read.py

```diff
@@ -456,11 +458,22 @@ def vars(self):
         """
 
         if not hasattr(self, "_read_vars"):
-            self._read_vars = Variables(
-                "file", path=self.filelist[0], product=self.product
-            )
-
+            self._read_vars = Variables(path=self.filelist[0])
         return self._read_vars
+    
+    @property
+    def filelist(self):
+        """
+        Return the list of files represented by this Read object.
+        """
+        return self._filelist
+
+    @property
+    def product(self):
+        """
+        Return the product associated with the Read object.
+        """
+        return self._product
```

Comment:
more duplicate code

Reply:
So I would argue that this code isn't duplicate. Instead it is making a private attribute accessible as a read-only public attribute to the user. The user _could_ technically access and modify `_filelist` or `_product`, but the underscore warns them that they really shouldn't be doing that. Instead we have provided publicly available `filelist` and `product` which the user can see, but can't modify, since we don't want them changing that information.